client/buffered_client.go

```diff
@@ -0,0 +1,70 @@
+package client
+
+import "time"
+
+type BufferConfig struct {
+	FlushMaxPoints   int
+	FlushMaxWaitTime time.Duration
+}
+
+func NewBufferedClient(clientConfig Config, bufferConfig BufferConfig) (bufferedClient *BufferedClient, err error) {
+	client, err := NewClient(clientConfig)
+	if err != nil {
+		return
+	}
+	bufferedClient = &BufferedClient{
+		Client:       client,
+		bufferConfig: bufferConfig,
+		ingestChan:   make(chan Point, bufferConfig.FlushMaxPoints/3),
+		flushTimer:   time.NewTimer(bufferConfig.FlushMaxWaitTime),
+		pointsBuf:    make([]Point, bufferConfig.FlushMaxPoints),
+		pointsIndex:  0,
+	}
+	go bufferedClient.ingestAndFlushLoop()
```

Comment:
That's a reasonable need :) Will have to add a `FlushAndStop()` which flushes outstanding points and closes the ingest channel (thus preventing further calls to `Add()`